server/src/core/search/search_executor/meilisearch.rs

```diff
@@ -0,0 +1,163 @@
+use awc::Client;
+
+use super::preprocess;
+use serde::{Deserialize, Serialize};
+use serde_json::Result;
+use std::collections::HashMap;
+
+/// Input into MeiliSearch
+pub(super) struct MSSearchArgs<'a> {
+    pub(super) q: &'a str,
+    pub(super) filter: Option<MSSearchFilter>,
+    pub(super) limit: usize,
+}
+
+#[derive(Debug)]
+pub(super) struct MSSearchFilter {
+    facet: Vec<String>,
+}
+
+#[derive(Serialize)]
+struct MSQuery<'a> {
+    q: &'a str,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    filter: Option<Vec<Vec<String>>>,
+    limit: usize,
+    #[serde(rename = "facetsDistribution")]
+    facets_distribution: Vec<String>,
+}
+
+/// Result format of MeiliSearch.
+#[derive(Deserialize)]
+#[allow(dead_code)]
+pub(super) struct MSResults {
+    pub(super) hits: Vec<MSHit>,
+    #[serde(rename = "nbHits")]
+    pub(super) nb_hits: i32,
+    #[serde(rename = "facetsDistribution")]
+    pub(super) facets_distribution: MSFacetDistribution,
+}
+
+#[derive(Deserialize, Debug)]
+pub(super) struct MSFacetDistribution {
+    pub(super) facet: HashMap<String, i32>,
+}
+
+#[derive(Deserialize, Clone)]
+#[allow(dead_code)]
+pub(super) struct MSHit {
+    ms_id: String,
+    pub(super) id: String,
+    pub(super) name: String,
+    pub(super) arch_name: Option<String>,
+    pub(super) r#type: String,
+    pub(super) type_common_name: String,
+    pub(super) parent_building: Vec<String>,
+    parent_keywords: Vec<String>,
```

Comment:
that only parts of these are marked public is, because rust was happy and I did not want to make unnessesarily many things public. What do you think about this?

Reply:
It's probably because `parent_keywords` (which basically contains the ids of the parent buildings / sites) is included in the responses (it's mainly for MeiliSearch, so I think it's fine if only this one remains private).